### PR TITLE
[feat]: [CI-7622]: Add thread dump on log service pod restart

### PR DESCRIPTION
**Pull request by smjt-h**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46122)

### CODE REVIEW

File: product/log-service/main.go [hunk 1]
```diff
@@ -29,5 +33,32 @@ func main() {
...
+	podName := os.Getenv("POD_NAME")
+	// Write the stack trace to a file or log it
+	f, err := os.Create(podName + "thread_dump.txt")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	if _, err := f.Write(buf); err != nil {
+		log.Fatal(err)
```
Comment (vistaarjuneja) [inline product/log-service/main.go:61]: Ditto, don't use log.Fatal

File: product/log-service/main.go [hunk 2]
```diff
@@ -29,5 +33,32 @@ func main() {
...
 	}()
 
+	go threadDump()
+
 	cli.Command()
 }
+
+func threadDump() {
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt, os.Kill, syscall.SIGUSR1)
```
[the remark this answers is not in the document]
Comment (smjt-h) [reply product/log-service/main.go]: added sigkill and sigterm as well

File: product/log-service/main.go [hunk 3]
```diff
@@ -29,5 +33,32 @@ func main() {
...
+
+	// Get a stack trace for each goroutine
+	buf := make([]byte, 1<<16)
+	length := runtime.Stack(buf, true)
+
+	podName := os.Getenv("POD_NAME")
+	// Write the stack trace to a file or log it
+	f, err := os.Create(podName + "thread_dump.txt")
+	if err != nil {
+		log.Fatal(err)
```
[the remark this answers is not in the document]
Comment (smjt-h) [reply product/log-service/main.go]: done

File: product/log-service/main.go (hunk 1 above)
Comment (smjt-h) [reply product/log-service/main.go]: done

**Comment by smjt-h** [comment]
trigger gobuild

**Comment by smjt-h** [comment]
trigger utAll

**Comment by smjt-h** [comment]
trigger utAll

**Comment by smjt-h** [comment]
trigger uts

**Review by vistaarjuneja — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46122#-:-NV9Z1yJ8NXY8_6qpoM4:bnfp4nl)* status: Some of the checks are too old. 1 file(s) to review.

**Review by vistaarjuneja — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46122#-:-NV9Z9JL9Xx0C47RIIjG:bnfp4nl)* status: Some of the checks are too old.

**Pull request merged**